pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -526,6 +526,16 @@ func (lb *ListenerBuilder) buildInboundChainConfigs() []inboundChainConfig {
 	return chainConfigs
 }
 
+func findTargetPort(svcPort uint32, serviceInstances []*model.ServiceInstance) uint32 {
```

Comment:
I am not sure this makes sense. The Sidecar Ingress section is for explicitly overriding the ingress config. Implicitly overriding their port choices doesn't seem right.

What are you trying to do?

Reply:
The listener will refer to a cluster with the targetPort, if the port is using svcPort, it will block

Reply:
Look at https://github.com/istio/istio/pull/42482/files#diff-4afa570feba990275e28fa8ca0b5be122e9ff5d6f0beee9d2e8f54cb57d46613R489

Reply:
That link just sent back to this PR?

Reply:
I mean L489, inbound filter chain points to inbound cluster, only associate with targetPort. If the port we use is wrong, the result is blocking

			cc := inboundChainConfig{
				// Sidecar config doesn't have a real hostname. In order to give some telemetry info, make a synthetic hostname.
				telemetryMetadata: telemetry.FilterChainMetadata{
					InstanceHostname: host.Name(lb.node.SidecarScope.Name + "." + lb.node.SidecarScope.Namespace),
				},
				port:        port,
		              **clusterName: model.BuildInboundSubsetKey(int(port.TargetPort)),**
				bind:        i.Bind,
				bindToPort:  bindtoPort,
			}

Reply:
I mean what is the user facing behavior you are trying to solve?

Given there was no targetPort in the API for a long time, its no clear why it should be added now.

Reply:
When user want to set ingress tls setting, they will need to set ingress service port. However without target port, it default to service port and the traffic will goes to a non existing cluster.

Reply:
They should set the port number to the target, not the service port.

By the time the request gets to Envoy it's already resolved to targetPort.

Reply:
Ic, i find a bug cause the confusion to me. Will fix it